checks/cs50/2017/x/speller/checks.py

```diff
@@ -0,0 +1,58 @@
+import os
+import sys
+
+sys.path.append(os.getcwd())
+from check50 import *
+
+class Speller(Checks):
+
+    @check()
+    def exists(self):
+        """dictionary.c, dictionary.h, and Makefile exist"""
+        super(Speller, self).exists("dictionary.c")
+        super(Speller, self).exists("dictionary.h")
+        super(Speller, self).exists("Makefile")
+
+    @check("exists")
+    def compiles(self):
+        """speller compiles"""
+        self.include("speller.c")
+        self.spawn("make").exit(0)
+
+    @check("compiles")
+    def basic(self):
+        """handles most basic words properly"""
+        self.include("basic")
+        self.spawn("./speller basic/dict basic/text").stdout(File("basic/out")).exit(0)
```

Comment:
We do need to be a descendant of `unittest.TestCase` otherwise `unittest` itself won't know which checks to run: just storing the instance as a private instance variable means that the instance wouldn't contain any checks to run, since checks are defined as methods of the `unittest.TestCase` subclass.